### PR TITLE
Add free course on Spark

### PR DESCRIPTION
**Pull request by DrishtiJ**
## What does this PR do?
Add resource

### Description
Spark Course

### Why is this valuable (or not)?
Learn Spark and also Machine Learning with Spark

### How do we know it's really free?
Listed as free course on Udemy

### For book lists, is it a book? For course lists, is it a course? etc.
Course

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: courses/free-courses-en.md [hunk 1]
```diff
@@ -947,6 +947,7 @@
 
 ### Spark
 
+* [Learn Spark](https://www.udacity.com/course/learn-spark-at-udacity--ud2002) - by INSIGHT on Udemy
```
Comment (davorpa) [inline courses/free-courses-en.md:950]: Fix and completing metadata

```suggestion
* [Learn Spark](https://www.udacity.com/course/learn-spark-at-udacity--ud2002) - INSIGHT, `tch.:` David Drummond, `tch.:` Judit Lantos (Udacity)
```

**Review by davorpa — APPROVED** [review]

File: courses/free-courses-en.md [hunk 2]
```diff
@@ -947,6 +947,7 @@
 
 ### Spark
 
+* [Learn Spark](https://www.udacity.com/course/learn-spark-at-udacity--ud2002) - INSIGHT, `tch.:` David Drummond, `tch.:` Judit Lantos (Udacity)
```
Comment (eshellman) [inline courses/free-courses-en.md:950]:
```suggestion
* [Learn Spark](https://www.udacity.com/course/learn-spark-at-udacity--ud2002) -  David Drummond, Judit Lantos (Udacity)
```
sorry for the back and forth - we're still figururing out the best way to denote contributor roles. In this case "Insight" is a sponsor or publisher more than a contributor, which we usually don't list.

Comment (DrishtiJ) [reply courses/free-courses-en.md]: Sure, committed the change.

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Pull request merged**